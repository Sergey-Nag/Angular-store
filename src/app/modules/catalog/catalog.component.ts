import { Component, OnInit } from '@angular/core';
import { HttpLoadingService } from '@core/services/http-loading.service';

@Component({
  selector: 'app-catalog',
  templateUrl: './catalog.component.html',
  styleUrls: ['./catalog.component.scss']
})
export class CatalogComponent {
}
