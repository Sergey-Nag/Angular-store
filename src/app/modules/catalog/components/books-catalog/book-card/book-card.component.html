<div class="card border-light">
    <img [src]="book.cover" class="card-img-top" alt="...">
    <div class="card-body">
      <h5 class="card-title">{{ book.title }}</h5>
      <p class="card-text">{{ book.author }}</p>
      <div class="row">
          <div class="col-6">
              <span class="price">${{ book.price }}</span>
          </div>
          <div class="col-6 d-grid">
            <a href="#" class="btn btn-outline-primary">View</a>
          </div>
      </div>
    </div>
  </div>