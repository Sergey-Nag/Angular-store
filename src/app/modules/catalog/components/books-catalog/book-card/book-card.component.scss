.price {
    font-size: 1.4em;
    line-height: 1.7em;
    color: var(--bs-success);
}