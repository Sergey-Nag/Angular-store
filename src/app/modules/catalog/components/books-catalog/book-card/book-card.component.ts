import { ChangeDetectionStrategy, Component, Input, OnInit } from '@angular/core';
import { Book } from 'src/app/shared/book.model';

@Component({
  selector: 'app-book-card',
  templateUrl: './book-card.component.html',
  styleUrls: ['./book-card.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class BookCardComponent {
  @Input() book!: Book;
  
  constructor() { }

}
