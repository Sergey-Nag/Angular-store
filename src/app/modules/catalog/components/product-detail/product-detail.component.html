<div class="row">
    <div class="col-md-8">
        <app-product-info></app-product-info>
    </div>
    <div class="col-md-4 mb-5">
        <app-product-calc></app-product-calc>
        <button class="btn btn-primary float-end mt-4" disabled>Add to cart</button>
    </div>
</div>