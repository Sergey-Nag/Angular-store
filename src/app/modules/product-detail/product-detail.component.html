<div class="row" *ngIf="book">
    <div class="col-md-8">
        <app-product-info [book]="book"></app-product-info>
    </div>
    <div class="col-md-4 mb-5">
        <app-product-calc
            [price]="book.price"
            [available]="book.count"
            (onCountChange)="handleCountChange($event)">
        </app-product-calc>
        <button
            class="btn btn-primary float-end mt-4"
            [disabled]="count === 0">
            Add to cart
        </button>
    </div>
</div>
<app-spinner *ngIf="isLoading$ | async"></app-spinner>