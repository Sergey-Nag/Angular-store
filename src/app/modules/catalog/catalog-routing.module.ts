import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { CartComponent } from 'src/app/components/cart/cart.component';
import { AuthGuard } from '../auth/guards/auth.guard';
import { CatalogComponent } from './catalog.component';
import { BooksCatalogComponent } from './components/books-catalog/books-catalog.component';
import { ProductDetailComponent } from './components/product-detail/product-detail.component';

const routes: Routes = [
  { path: '', pathMatch: 'full', component: BooksCatalogComponent },
  { path: ':id', component: ProductDetailComponent }
]
@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule]
})
export class CatalogRoutingModule { }