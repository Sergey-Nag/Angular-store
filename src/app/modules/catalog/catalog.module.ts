import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { CatalogRoutingModule } from './catalog-routing.module';
import { BooksCatalogComponent } from './components/books-catalog/books-catalog.component';
import { ProductDetailComponent } from './components/product-detail/product-detail.component';
import { ProductCalcComponent } from './components/product-detail/product-calc/product-calc.component';
import { ProductInfoComponent } from './components/product-detail/product-info/product-info.component';
import { BooksFilterComponent } from './components/books-catalog/books-filter/books-filter.component';
import { BooksListComponent } from './components/books-catalog/books-list/books-list.component';
import { BookCardComponent } from './components/books-catalog/book-card/book-card.component';
import { FilterArrayPipe } from 'src/app/shared/pipes/filterArray.pipe';

@NgModule({
  declarations: [
    BooksCatalogComponent,
    ProductDetailComponent,
    ProductInfoComponent,
    ProductCalcComponent,
    BooksFilterComponent,
    BooksListComponent,
    BookCardComponent,
    FilterArrayPipe,
  ],
  imports: [
    CommonModule,
    CatalogRoutingModule,
    FormsModule,
  ],
})
export class CatalogModule { }
