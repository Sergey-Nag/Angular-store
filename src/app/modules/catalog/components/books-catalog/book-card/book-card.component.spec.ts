import { APP_BASE_HREF, Location } from "@angular/common";
import { ComponentFixture, fakeAsync, TestBed, tick } from "@angular/core/testing";
import { Book } from "@shared/models/book.model";
import { BookCardComponent } from "./book-card.component";
import { render, screen } from '@testing-library/angular'
import { HttpClientTestingModule } from "@angular/common/http/testing";
import { AppRoutingModule } from "src/app/app-routing.module";
import { RouterTestingModule } from '@angular/router/testing';
import { CatalogModule } from "../../../catalog.module";
import { Router } from "@angular/router";
import { ProductDetailComponent } from "src/app/modules/product-detail/product-detail.component";

describe('Book card', () => {
  let book: Book;
  let component: BookCardComponent;
  let fixtureComponent: ComponentFixture<BookCardComponent>;
  let router: Router;

  beforeEach(async () => {
    book = new Book(
      '1',
      6, 
      56.95,
      'Apuntes de Javascript I - Nivel Intermedio',
      'JuanMa Garrido', 
      'Intermediate', 
      'En Castellano) Revision de conceptos (actuales) de javascript desde basicos hasta un nivel intermedio',
      'https://jsbooks.revolunet.com/img/cover-apuntes-javascript-intermedio.png',
      ['core']
    );

    const { fixture } = await render(BookCardComponent, {
      componentProperties: { book },
      imports: [RouterTestingModule.withRoutes(
        [{
          path: '',
          children: [
            {
              path: ':id',
              component: ProductDetailComponent,
            }
          ]
        }]
      ), CatalogModule, HttpClientTestingModule],
    });

    fixtureComponent = fixture;
    component = fixture.componentInstance;
    router = TestBed.inject(Router);
  });

  it('Should show the book content', () => {
    expect(fixtureComponent).toMatchSnapshot();
  });

  it('Should go to the details page by clicking on the view link', fakeAsync(() => {
    screen.getByRole<HTMLLinkElement>('link', { name: 'View' }).click();
    
    tick();
    
    expect(router.routerState.snapshot.url).toBe('/1')
  }));
});