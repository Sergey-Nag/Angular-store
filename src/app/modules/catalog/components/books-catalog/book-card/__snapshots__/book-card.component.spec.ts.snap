// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book card Should show the book content 1`] = `
<app-book-card
  __ngContext__={[Function LRootView]}
>
  <div
    class="card border-light"
  >
    <img
      alt="..."
      class="card-img-top"
      src="https://jsbooks.revolunet.com/img/cover-apuntes-javascript-intermedio.png"
    />
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Apuntes de Javascript I - Nivel Intermedio
      </h5>
      <p
        class="card-text"
      >
        JuanMa Garrido
      </p>
      <div
        class="row"
      >
        <div
          class="col-6"
        >
          <span
            class="price"
          >
            $56.95
          </span>
        </div>
        <div
          class="col-6 d-grid"
        >
          <a
            class="btn btn-outline-primary"
            href="/1"
          >
            View
          </a>
        </div>
      </div>
    </div>
  </div>
</app-book-card>
`;
